Change dependencies to compileOnly so they are not in POM or Gradle Module

jamesfredley opened the pull request:
Switched several dependencies from implementation/api to compileOnly. Disabled Grails springDependencyManagement and added testImplementation for grails-bom.

Switched to `grails-dependencies-starter-web` and `grails-dependencies-test` dependencies to simplify

resolves:  https://github.com/apache/grails-quartz/issues/158

jamesfredley committed:
Change dependencies to compileOnly so they are not in POM or Gradle Module

Switched several dependencies from implementation/api to compileOnly. Disabled Grails springDependencyManagement and added testImplementation for grails-bom.

jamesfredley commented:
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <!-- This module was also published with a richer model, Gradle metadata,  -->
  <!-- which should be used instead. Do not delete the following line which  -->
  <!-- is to indicate to Gradle or any Gradle module metadata file consumer  -->
  <!-- that they should prefer consuming it instead. -->
  <!-- do_not_remove: published-with-gradle-metadata -->
  <modelVersion>4.0.0</modelVersion>
  <groupId>org.apache.grails</groupId>
  <artifactId>grails-quartz</artifactId>
  <version>4.0.0-SNAPSHOT</version>
  <name>Grails Quartz Plugin</name>
  <description>This plugin allows your Grails application to schedule jobs to be executed using a specified interval or cron expression.</description>
  <url>https://github.com/apache/grails-quartz</url>
  <licenses>
    <license>
      <name>The Apache Software License, Version 2.0</name>
      <url>https://www.apache.org/licenses/LICENSE-2.0.txt</url>
      <distribution>repo</distribution>
    </license>
  </licenses>
  <developers>
    <developer>
      <id>burtbeckwith</id>
      <name>Burt Beckwith</name>
    </developer>
    <developer>
      <id>jeffscottbrown</id>
      <name>Jeff Scott Brown</name>
    </developer>
    <developer>
      <id>graemerocher</id>
      <name>Graeme Rocher</name>
    </developer>
    <developer>
      <id>ryanvanderwerf</id>
      <name>Ryan Vanderwerf</name>
    </developer>
    <developer>
      <id>sergeynebolsin</id>
      <name>Sergey Nebolsin</name>
    </developer>
    <developer>
      <id>puneetbehl</id>
      <name>Puneet Behl</name>
    </developer>
    <developer>
      <id>vitaliisamolovskikh</id>
      <name>Vitalii Samolovskikh</name>
    </developer>
  </developers>
  <scm>
    <connection>scm:git@github.com:apache/grails-quartz.git</connection>
    <developerConnection>scm:git@github.com:apache/grails-quartz.git</developerConnection>
    <url>https://github.com/apache/grails-quartz</url>
  </scm>
  <issueManagement>
    <system>GitHub Issues</system>
    <url>https://github.com/apache/grails-quartz/issues</url>
  </issueManagement>
  <dependencies>
    <dependency>
      <groupId>org.quartz-scheduler</groupId>
      <artifactId>quartz</artifactId>
      <version>2.5.0</version>
      <scope>compile</scope>
    </dependency>
  </dependencies>
</project>

```

jamesfredley committed:
Update Gradle wrapper to 8.14.3

Bump Gradle wrapper version from 8.14.2 to 8.14.3 to use the latest features and fixes.

jdaugherty commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.14.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.14.3-bin.zip
```
Gradle version is also in .sdkmanrc

jamesfredley replied:
I wish there was a way for .sdkmanrc to auto update.  I pushed another commit.

jamesfredley committed:
Update Gradle version to 8.14.3 in .sdkmanrc

Bumps the Gradle version from 8.14.2 to 8.14.3 for local development environment configuration.

jamesfredley committed:
Refactor dependencies in build.gradle

Reorganized and simplified dependency declarations by removing individual compileOnly and testImplementation entries in favor of 'grails-dependencies-starter-web' and 'grails-dependencies-test'. This streamlines dependency management and reduces redundancy.

jdaugherty commented on build.gradle:
```diff
@@ -57,48 +57,25 @@ repositories {
-    }
+    compileOnly platform("org.apache.grails:grails-bom:$grailsVersion")
 
-    implementation 'org.apache.groovy:groovy-sql', {
```
This is sitll used by JDBCCleanup

jamesfredley commented on build.gradle:
```diff
@@ -57,48 +57,25 @@ repositories {
-    implementation platform("org.apache.grails:grails-bom:$grailsVersion")
+dependencies {
 
     api "org.quartz-scheduler:quartz:$quartzVersion", {
```
`org.quartz-scheduler:quartz` version can be supplied from `grails-bom` via `spring-boot-dependencies`, but we would have to add: 

`api platform("org.apache.grails:grails-bom:$grailsVersion")`

or consolidate in the grails-bom dependencies to: 

`implementation platform("org.apache.grails:grails-bom:$grailsVersion")`

I believe

jamesfredley committed:
Refactor quartz dependencies to use Grails BOM platform

Updated build.gradle to use implementation platform for Grails BOM and removed redundant platform declarations. Also removed the unused quartzVersion property from gradle.properties, as the version is now managed by the BOM.

jdaugherty reviewed (approved):
Separately from this PR: we should open a ticket to create an example project to better test this going forward.  We can restructure the repo at the same time.
